eras/babbage/test-suite/src/Test/Cardano/Ledger/Babbage/Translation/TranslatableGen.hs

```diff
@@ -0,0 +1,163 @@
+
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Alonzo.Scripts (AlonzoScript (..), ExUnits (..), Tag (..))
+import Cardano.Ledger.Alonzo.Scripts.Data (Data (..))
+import Cardano.Ledger.Alonzo.Tx (AlonzoTx (..))
+import Cardano.Ledger.Alonzo.TxWits
+import Cardano.Ledger.Babbage.TxBody (BabbageTxBody (..), BabbageTxOut (..), Datum (..))
+import Cardano.Ledger.Binary (mkSized)
+import Cardano.Ledger.Credential (StakeReference (..))
+import Cardano.Ledger.Language (Language (..))
+import Cardano.Ledger.TxIn (TxIn (..))
+import Cardano.Ledger.UTxO (UTxO (..))
+import qualified Data.Map as Map
+import Data.Maybe.Strict
+import Data.Sequence.Strict (fromList)
+import qualified Data.Set as Set
+import qualified PlutusLedgerApi.V1 as PV1
+import Test.Cardano.Ledger.Alonzo.Translation.TranslationInstanceGen (TranslatableGen (..))
+import Test.Cardano.Ledger.Core.Arbitrary ()
+import Test.QuickCheck (
+  Arbitrary,
+  Gen,
+  arbitrary,
+  frequency,
+  listOf1,
+  oneof,
+  scale,
+  vectorOf,
+ )
+
+instance TranslatableGen Babbage where
+  tgTx l = genTx @Babbage l (genTxBody l)
+  tgUtxo = utxoWithTx @Babbage
+
+utxoWithTx ::
+  forall era.
+  ( EraTx era
+  , Arbitrary (Value era)
+  , Arbitrary (Script era)
+  , BabbageEraTxBody era
+  , TxOut era ~ BabbageTxOut era
+  ) =>
+  Language ->
+  Tx era ->
+  Gen (UTxO era)
+utxoWithTx l tx = do
+  let ins = tx ^. bodyTxL ^. inputsTxBodyL
+  let refIns = tx ^. bodyTxL ^. referenceInputsTxBodyL
+  let allIns = ins `Set.union` refIns
+  outs <- vectorOf (length allIns) (genTxOut @era l)
+  pure $ UTxO (Map.fromList $ Set.toList allIns `zip` outs)
+
+genTx ::
+  forall era.
+  ( Arbitrary (TxAuxData era)
+  , Arbitrary (Script era)
+  , AlonzoEraTxWits era
+  , AlonzoScript era ~ Script era
+  , AlonzoTxWits era ~ TxWits era
+  ) =>
+  Language ->
+  Gen (TxBody era) ->
+  Gen (AlonzoTx era)
+genTx l txbGen =
+  AlonzoTx
+    <$> txbGen
+    <*> genTxWits @era l
+    <*> arbitrary
+    <*> arbitrary
+
+genTxOut ::
+  forall era.
+  ( EraTxOut era
+  , Arbitrary (Value era)
+  , Arbitrary (Script era)
+  ) =>
+  Language ->
+  Gen (BabbageTxOut era)
+genTxOut l = do
+  addr <- genNonByronAddr @(EraCrypto era)
+  value <- scale (`div` 15) arbitrary
+  script <- case l of
+    PlutusV1 -> pure SNothing
+    _ -> arbitrary
+  datum <- case l of
+    PlutusV1 -> oneof [pure NoDatum, DatumHash <$> (arbitrary :: Gen (DataHash (EraCrypto era)))]
+    _ -> arbitrary
+  pure $ BabbageTxOut addr value datum script
+
+genTxWits ::
+  forall era.
+  ( Arbitrary (Script era)
+  , AlonzoEraTxWits era
+  , AlonzoScript era ~ Script era
+  , AlonzoTxWits era ~ TxWits era
+  ) =>
+  Language ->
+  Gen (AlonzoTxWits era)
+genTxWits = \case
+  PlutusV1 -> arbitrary
+  _ -> do
+    arbWits <- arbitrary :: Gen (AlonzoTxWits era)
+    pure $
+      arbWits
+        & rdmrsTxWitsL @era
+          .~ Redeemers (Map.singleton (RdmrPtr Spend 0) (Data (PV1.I 42), ExUnits 5000 5000))
```

Comment:
This looks odd. How come a single `Redeemer` is hard coded for PlutusV2 and PlutusV3?

Reply:
Mhm. It's because in v1 this is not checked, so I chose the easiest way to make the transaction validate. 
But it's perhaps too lazy of me, let me try to generate data instead of this

Reply:
I improved this slightly, with the least amount of hardcoding that allows us to not add more complexity to the generation of the txbody. This is because the redeemers have a relationship with the contents of the body. 
Since this is validated in the rule anyway, I thought it's not important to be so accurate about generating data for these tests.